lib/puppet/defaults.rb

```diff
@@ -1290,13 +1298,16 @@ def self.valid_file_checksum_types
     :rest_authconfig => {
       :default    => "$confdir/auth.conf",
       :type       => :file,
+      :deprecated  => :completely,
       :desc       => "The configuration file that defines the rights to the different
-      rest indirections.  This can be used as a fine-grained
-      authorization system for `puppet master`.",
+      rest indirections.  This can be used as a fine-grained authorization system for
+      `puppet master`.  The `puppet master` command is deprecated and Puppet Server
+      uses its own auth.conf that must be placed within its configuration directory.",
```

Comment:
Will puppetserver still use `:rest_authconf` if `use-legacy-auth-conf` is enabled?

Reply:
Yes, it will for routes that are still implemented using Puppet's Ruby rest interface. That excludes the CA and admin APIs which are in clojure (and anything else we move to clojure). In Puppet Server 5 `use-legacy-auth-conf` is [off by default](https://puppet.com/docs/puppetserver/5.3/config_file_auth.html#aside-changes-to-authorization-in-puppet-server-500)(docs) and [warns if enabled](https://github.com/puppetlabs/puppetserver/blob/master/src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj#L279-L281)(code).